File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/Notification.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;

/**
 *
 * @author simon
 * @since 2024/11/4 */
@Data
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class Notification {
    private String id;
    private Repository repository;
    private Subject subject;
    private String reason;
    private boolean unread;
    
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
    
    @JsonProperty("last_read_at")
    private LocalDateTime lastReadAt;
    
    private String url;
    
    @JsonProperty("subscription_url")
    private String subscriptionUrl;

    @Data
    public static class Repository {
        private Long id;
        
        @JsonProperty("node_id")
        private String nodeId;
        
        private String name;
        
        @JsonProperty("full_name")
        private String fullName;
        
        private Owner owner;
        private boolean isPrivate;
        
        @JsonProperty("html_url")
        private String htmlUrl;
        
        private String description;
    }

    @Data
    public static class Owner {
        private String login;
        private Long id;
        
        @JsonProperty("node_id")
        private String nodeId;
        
        @JsonProperty("avatar_url")
        private String avatarUrl;
        
        @JsonProperty("html_url")
        private String htmlUrl;
        
        private String type;
        
        @JsonProperty("site_admin")
        private boolean siteAdmin;
    }

    @Data
    public static class Subject {
        private String title;
        private String url;
        
        @JsonProperty("latest_comment_url")
        private String latestCommentUrl;
        
        private String type;
    }
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/Notification.kt
package com.cpy3f2.gixor_mobile.model.entity

import java.time.LocalDateTime

data class Notification(
    val id: String? = null,
    val repository: Repository? = null,
    val subject: Subject? = null,
    val reason: String? = null,
    val unread: Boolean = false,

    val updatedAt: LocalDateTime? = null,

    val lastReadAt: LocalDateTime? = null,
    
    val url: String? = null,

    val subscriptionUrl: String? = null
) {
    data class Repository(
        val id: Long? = null,

        val nodeId: String? = null,
        
        val name: String? = null,

        val fullName: String? = null,
        
        val owner: Owner? = null,
        val isPrivate: Boolean = false,

        val htmlUrl: String? = null,
        
        val description: String? = null
    )

    data class Owner(
        val login: String? = null,
        val id: Long? = null,

        val nodeId: String? = null,

        val avatarUrl: String? = null,

        val htmlUrl: String? = null,
        
        val type: String? = null,

        val siteAdmin: Boolean = false
    )

    data class Subject(
        val title: String? = null,
        val url: String? = null,

        val latestCommentUrl: String? = null,
        
        val type: String? = null
    )
}